frontend/public/components/utils/simple-tab-nav.tsx

```diff
@@ -43,6 +43,15 @@ export class SimpleTabNav extends React.Component<SimpleTabNavProps, SimpleTabNa
     });
   }
 
+  componentDidUpdate(prevProps) {
+    if (!_.isEqual(this.props?.tabProps?.item?.obj, prevProps?.tabProps?.item?.obj)) {
+      const { tabs, selectedTab: tabName } = this.props;
+      this.setState({
+        selectedTab: _.find(tabs, { name: tabName }) || _.head(tabs),
+      });
+    }
+  }
```

Comment:
This should be implemented as `getDerivedStateFromProps`.
Have the `state.selectedTab` be just the tab name and NOT the tab object which also contains other info.
Inside the render function, loop through the `props.tabs` to find the correct tab to render. This way we don't store stale data in the state.